modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -137,6 +141,11 @@ template<typename Net> class Params {
         return *this;
     }
 
+    Params<Net>& cfgInputReshape(const std::map<std::string, std::vector<std::size_t>> &r) {
+        desc.reshape_inputs = r;
+        return *this;
+    }
```

Comment:
Really I'd add an overload with `const std::string, const std::vector<size_t>`.

Reply:
@mpashchenkov what about this?

Reply:
Consider to add `rvalue` implementation, because it will be used in most of cases

Reply:
Added overloads.